src/commands/HBonds.cpp

```diff
@@ -0,0 +1,238 @@
+            options.angle = std::stod(args.at("--angle").asString())*pi/180;
+        } catch(std::invalid_argument) {
+            throw CFilesError("Invalid argument for angle criterion");
+        }
+    }	
+
+    return options;
+}
+
+
+std::string HBonds::description() const {
+    return "compute hydrogen bonds network";
+}
+
+int HBonds::run(int argc, const char* argv[]) {
+    auto options = parse_options(argc, argv);
+
+    selection_acceptor_ = Selection(options.selection_acceptor);
+    if (selection_acceptor_.size() != 2) {
+        throw CFilesError("Can not use a selection for acceptors with size different than 2.");
+    }
+
+    auto selection_string = split(selection_acceptor_.string(),':');
+    if (selection_string[0] != "bonds") {
+        throw CFilesError("'Bonds' type selection compulsory for acceptors.");
+    }
+
+    selection_donor_ = Selection(options.selection_donor);
+    if (selection_donor_.size() != 1) {
+        throw CFilesError("Can not use a selection for donors with size larger than 1.");
+    }
+
+    auto infile = Trajectory(options.trajectory, 'r', options.format);
+    std::ofstream outfile(options.outfile, std::ios::out);
+    if (outfile.is_open()) {
+        outfile << "#Hydrogen bond network in trajectory " << options.trajectory << std::endl;
+        outfile << "# Selection: acceptors: " << options.selection_acceptor;
+        outfile << " and donors: " << options.selection_donor << std::endl;
+        outfile << "# Criteria:" << std::endl;
+        outfile << "# donor-acceptor distance < " << options.distance << " angstroms" << std::endl;
+        outfile << "# donor-acceptor-H angle < " << options.angle*180/pi << " degrees" << std::endl;
+    } else {
+        throw CFilesError("Could not open the '" + options.outfile + "' file.");
+    }
+
+    if (options.custom_cell) {
+        infile.set_cell(options.cell);
+    }
+
+    if (options.topology != "") {
+        infile.set_topology(options.topology, options.topology_format);
+    }
+
+    for (auto step: options.steps) {
+        if (step >= infile.nsteps()) {
+            break;
+        }
+        auto frame = infile.read_step(step);
+        if (options.guess_bonds) {
+            frame.guess_topology();
+        }
+
+        outfile << "# Frame: " << step << std::endl;        
+
+        auto positions = frame.positions();
+        auto cell = frame.cell();
+
+        auto matched = selection_acceptor_.evaluate(frame);
+        for (auto match: matched) {
+            assert(match.size() == 2);
+
+            size_t acceptor = static_cast<size_t>(-1);
+            size_t hydrogen = static_cast<size_t>(-1);
+
+            if (frame.topology()[match[0]].type() == "H") {
+                acceptor = match[1];
+                hydrogen = match[0];
+            }
+            else if (frame.topology()[match[1]].type() != "H") {
+                acceptor = match[0];
+                hydrogen = match[1];
+            }
+            else
```

Comment:
Could you write this `} else {` in a single line to follow the rest of the code ?